packages/vda-did-resolver/src/getDIDs.ts

```diff
@@ -28,13 +28,21 @@ export async function getDIDs(network: string, startIndex: number = 0, count: nu
     const contract = new Contract(address, contractABI.abi, provider);
 
     if (mostRecent) {
-        const activeDids = await activeDIDCount(network)
-        startIndex = activeDids - startIndex - count
+        // user wants most recent DIDs
+        // we need to calculate the correct offset
+        // Note that startIndex is indexed from the **end** of the list of DIDs
+        //   so a startIndex of 0 and count of 20 means get the 20 most recent.
+        const activeDidCount = await activeDIDCount(network)
+        startIndex = activeDidCount - startIndex - count
     }
     
-    let data;
+    let data: string[] | undefined = undefined;
     try {
-        data = await contract.callStatic.getDIDs(startIndex, count)
+        data = (await contract.callStatic.getDIDs(startIndex, count)) as string[];
+        data.map((did) => {
+            // map the raw string into the correct canonoical DID format
+            return `did:vda:${network}:${did}`;
```

Comment:
Do we have a centralised util function for this formatting?

Reply:
Not that I've seen anywhere